src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/LuceneDocumentConverterBase.cs

```diff
@@ -38,7 +38,14 @@ public abstract class LuceneDocumentConverterBase : IDisposable
 
         protected LuceneDocumentConverterBase(ICollection<IndexField> fields, bool reduceOutput = false)
         {
-            _fields = fields.ToDictionary(x => x.Name, x => x, StringComparer.OrdinalIgnoreCase);
+            _fields = fields.ToDictionary(x => x.Name, x =>
```

Comment:
Yes, that is a problem.
Consider the case where we store a number. When we read it back from the lucene stored fields, we don't know it is a number, so we send it as a string, which can cause problems for some customers.
That is why we store the full json output in Lucene and send it directly, to preserve those kind of details.